pyogrio/geopandas.py

```diff
@@ -327,7 +344,12 @@ def write_dataframe(
     field_data = []
     field_mask = []
     for name in fields:
-        col = df[name].values
+        ser = df[name]
+        col = ser.values
+        if isinstance(ser.dtype, pd.DatetimeTZDtype):
+            # Deal with datetimes with timezones as strings
+            col = ser.astype(str)
```

Comment:
How is this sufficient? (then GDAL sees string values and will write it as strings instead of datetimes in the file format?) I think we still need to pass down to _io.pyx somehow that those fields are datetime, so we create the OGRField with the correct type.

Reply:
It's not - I have no idea why I thought this would work. 

I've come back to this now, and as far as I can see, I don't think we can parse strings down and ask GDAL to convert them - at least not with what we have exposed in pyogrio. I'm certainly not well across GDAL so happy to be corrected, but it seems we would need to call `OGRParseDate` do to this (I can't see any other way). But this operates on the `OGRField *psField`, whereas the pyogrio code operates on `OGRFeatureH` using the setters and getters to update values for fields without them being exposed directly.

So in the updated code I've pushed, it's using a numpy object array of datetimes to pass the timezone information down for each row. I did have a decent look in the pandas code to see if there's a direct way of extracting utc offsets, but as far as I can see it's not provided as a vectorised method.